IdentitySdkCore/IdentitySdkCore/Classes/ReachFive.swift

```diff
@@ -190,16 +190,65 @@ public class ReachFive: NSObject {
                 self.loginCallback(tkn: authenticationToken.tkn,scopes:scopes){ (authToken) -> Any in
                     _ = completion(authToken)
                 }
-        }
+            }
     }
     
     public func listWebAuthnDevices(authToken: AuthToken) -> Future<[DeviceCredential], ReachFiveError> {
-         
-           return self.reachFiveApi
+        
+        return self.reachFiveApi
             .getWebAuthnRegistrations(authorization: buildAuthorization(authToken: authToken))
-       }
+    }
     
     private func buildAuthorization (authToken: AuthToken) -> String {
         return authToken.tokenType! + " " + authToken.accessToken
     }
+    public func addNewWebAuthnDevice(authToken: AuthToken,
+                                     origin: String,
+                                     friendlyName: String?,viewController: UIViewController) -> Future<(), ReachFiveError>  {
+        
+        let thePromise = BrightFutures.Promise<(), ReachFiveError>()
+        let webAuthnRegistrationRequest = WebAuthnRegistrationRequest(
+            origin: origin,
+            friendlyName: friendlyName ?? ""
+        )
+        
+        self.reachFiveApi
+            .createWebAuthnRegistrationOptions(authorization: buildAuthorization(authToken: authToken),webAuthnRegistrationRequest: webAuthnRegistrationRequest)
+            .onSuccess{registrationOptions in
+                // prepare and setup the reachFiveClientFido
+                let reachFiveClientFido = ReachFiveFidoClient (viewController: viewController, origin: origin)
+                reachFiveClientFido.setupWebAuthnClient()
+                reachFiveClientFido.startRegistration(registrationOption: registrationOptions).onSuccess{ webauthnSignupCredential in
+                    // start the onSignupWithWebAuthnResult func to get firstly the authenticationToken and then exchange the tkn with an access token
+                    self.onAddNewWebAuthnDeviceResult(authToken: authToken, webauthnSignupCredential: webauthnSignupCredential)
+                        .onSuccess{
+                            let result: Swift.Result<(), ReachFiveError> = Swift.Result.success(())
+                            thePromise.complete(result)
+                        }
+                        .onFailure {error in
+                            thePromise.failure(error)
+                        }
+                }
+                .onFailure {error in
+                    thePromise.failure(error)                    
+                }
+            }
+            .onFailure {error in
+                thePromise.failure(error)
+            }
+        
+        return thePromise.future
```

Comment:
Why this pattern ? Why not use a combinator such as `flatMap` ? It would permit to have the same return without this intermediary `Promise` and the whole succession of `onFailure`/`onSuccess`.

Reply:
Thanks for this great note.